src/components/Pagination/Pagination.tsx

```diff
@@ -2,0 +1,92 @@
+import React from 'react';
+import cn from 'classnames';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+};
+
+const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  let totalPages = 0;
+
+  if (perPage !== 0) {
+    totalPages = Math.ceil(total / perPage);
+  }
+
+  const pagesList = Array.from({ length: totalPages }, (_, index) => index + 1);
+
+  const showPages = pagesList.map(page => (
+    <li
+      key={page}
+      className={cn('page-item', { active: currentPage === page })}
+    >
+      <a
+        data-cy="pageLink"
+        className="page-link"
+        href={`#${page}`}
+        onClick={() => onPageChange(page)}
+      >
+        {page}
+      </a>
+    </li>
+  ));
+
+  const handlePrevClick = (event: React.MouseEvent<HTMLAnchorElement>) => {
+    event.preventDefault();
+    if (currentPage > 1) {
+      onPageChange(currentPage - 1);
+    }
+  };
+
+  const handleNextClick = (event: React.MouseEvent<HTMLAnchorElement>) => {
+    event.preventDefault();
+    const lastPage = pagesList.length > 0 ? pagesList[pagesList.length - 1] : 1;
+
+    if (currentPage < lastPage) {
+      onPageChange(currentPage + 1);
+    }
```

Comment:
The logic to determine the last page is duplicated here and in line 73-75. You can calculate this once outside of the `handleNextClick` and use the value in both places to keep the code DRY (Don't Repeat Yourself).